plugins/module_utils/network/sonic/config/stp/stp.py

```diff
@@ -0,0 +1,1288 @@
+        requests = []
+
+        if not commands:
+            return requests
+
+        global_request = self.get_modify_stp_global_request(commands, have)
+        interfaces_request = self.get_modify_stp_interfaces_request(commands)
+        mstp_requests = self.get_modify_stp_mstp_request(commands, have)
+        pvst_request = self.get_modify_stp_pvst_request(commands)
+        rapid_pvst_request = self.get_modify_stp_rapid_pvst_request(commands)
+
+        if global_request:
+            requests.append(global_request)
+        if interfaces_request:
+            requests.append(interfaces_request)
+        if mstp_requests:
+            requests.append(mstp_requests)
+        if pvst_request:
+            requests.append(pvst_request)
+        if rapid_pvst_request:
+            requests.append(rapid_pvst_request)
+
+        return requests
+
+    def get_modify_stp_global_request(self, commands, have):
+        request = None
+
+        if not commands:
+            return request
+
+        stp_global = commands.get('global', None)
+        if stp_global:
+            global_dict = {}
+            config_dict = {}
+            enabled_protocol = stp_global.get('enabled_protocol', None)
+            loop_guard = stp_global.get('loop_guard', None)
+            bpdu_filter = stp_global.get('bpdu_filter', None)
+            disabled_vlans = stp_global.get('disabled_vlans', None)
+            root_guard_timeout = stp_global.get('root_guard_timeout', None)
+            portfast = stp_global.get('portfast', None)
+            hello_time = stp_global.get('hello_time', None)
+            max_age = stp_global.get('max_age', None)
+            fwd_delay = stp_global.get('fwd_delay', None)
+            bridge_priority = stp_global.get('bridge_priority', None)
+
+            if enabled_protocol:
+                config_dict['enabled-protocol'] = [stp_map[enabled_protocol]]
+            if loop_guard is not None:
+                config_dict['loop-guard'] = loop_guard
+            if bpdu_filter is not None:
+                config_dict['bpdu-filter'] = bpdu_filter
+            if disabled_vlans:
+                if have:
+                    cfg_stp_global = have.get('global', None)
+                    if cfg_stp_global:
+                        cfg_disabled_vlans = cfg_stp_global.get('disabled_vlans', None)
+                        if cfg_disabled_vlans:
+                            disabled_vlans = self.get_vlans_diff(disabled_vlans, cfg_disabled_vlans)
+                            if not disabled_vlans:
+                                commands['global'].pop('disabled_vlans')
+                if disabled_vlans:
+                    config_dict['openconfig-spanning-tree-ext:disabled-vlans'] = self.convert_vlans_list(disabled_vlans)
+            if root_guard_timeout:
+                config_dict['openconfig-spanning-tree-ext:rootguard-timeout'] = root_guard_timeout
+            if portfast is not None and enabled_protocol == 'pvst':
```

Comment:
Note I realize that subsequent discussion with Divya and experimentation with the current SONiC image may alter the context for this question, but I am posting it anyway in case it is still relevant.

Reply:
The SONiC code currently doesn't allow for any value of global portfast for mst and rapid-pvst protocols. For interfaces portfast, it allows the value to be false for mst and rapid-pvst protocols. 

I added a json fail for global for the reasons described above.